src/data/trainer-config.ts

```diff
@@ -1811,12 +1812,92 @@ export const trainerConfigs: TrainerConfigs = {
   [TrainerType.BAKER]: new TrainerConfig(++t)
     .setEncounterBgm(TrainerType.CLERK)
     .setMoneyMultiplier(1.35)
-    .setSpeciesFilter(s => s.isOfType(PokemonType.GRASS) || s.isOfType(PokemonType.FIRE)),
-  [TrainerType.BEAUTY]: new TrainerConfig(++t).setMoneyMultiplier(1.55).setEncounterBgm(TrainerType.PARASOL_LADY),
+    .setSpeciesFilter(
+      s =>
+        !![s.ability1, s.ability2, s.abilityHidden].find(
```

Comment:
```suggestion
        [s.ability1, s.ability2, s.abilityHidden].some(
```
In cases where you just want to know whether something exists but don't care about actually returning the value, you can use `.some()` instead of `.find()` so that it returns a boolean `true`/`false` (thus removing the need for the `!!` to force a `true`/`false` value).